### DISCUSSION TITLE
Exception on TracksRenderedChildren

### DISCUSSION THREAD

**Discussion by allgood** [livewire/livewire#3665] [7. Bugs (archived)] [answered]
I've stumbled on a situation very specific where the key of components was also using numeric sequences and the autogenerated ones, and, sometimes I have a component with a numeral and the id of the generated ones started with that numeral.

When debugging `TracksRenderedChildren` class I was able to pin this problem:

![image](https://user-images.githubusercontent.com/6070736/132223041-411c930e-588b-479b-b45f-6fa3f785c8e8.png)

When the method `childHasBeenRendered()` is called with the parameter 7, as a numeral, with the data above, it returns true when it should return false, causing an exception later in the code.

Looks like the solution is to enforce strict comparison on `childHasBeenRendered()` call on `in_array()` function.

I will submit a PR right now that fixes this problem.

**Comment by joshhanley** [comment, marked as answer]
Nice fix @allgood!

For anyone else, a fix has been merged in PR #3666
